spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/inbound/AbstractHazelcastMessageProducer.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.hazelcast.inbound;
+
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.util.EventObject;
+import java.util.HashSet;
+import java.util.Set;
+
+import com.hazelcast.core.AbstractIMapEvent;
+import com.hazelcast.core.DistributedObject;
+import com.hazelcast.core.EntryEvent;
+import com.hazelcast.core.EntryListener;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.MapEvent;
+import com.hazelcast.core.MultiMap;
+
+import reactor.util.StringUtils;
+
+import org.springframework.integration.endpoint.MessageProducerSupport;
+import org.springframework.integration.hazelcast.common.CacheEventType;
+import org.springframework.integration.hazelcast.common.CacheListeningPolicyType;
+import org.springframework.integration.hazelcast.common.HazelcastIntegrationDefinitionValidator;
+import org.springframework.integration.hazelcast.config.HazelcastLocalInstanceRegistrar;
+import org.springframework.util.Assert;
+
+/**
+ * Hazelcast Base Event-Driven Message Producer.
+ *
+ * @author Eren Avsarogullari
+ * @since 1.0.0
+ */
+public abstract class AbstractHazelcastMessageProducer extends MessageProducerSupport {
+
+	private final DistributedObject distributedObject;
+
+	private String cacheEventTypes;
+
+	private CacheListeningPolicyType cacheListeningPolicy;
+
+	private String hazelcastRegisteredEventListenerId;
+
+	private Set<String> cacheEvents;
+
+	protected AbstractHazelcastMessageProducer(DistributedObject distributedObject) {
+		Assert.notNull(distributedObject, "cache must not be null");
+		this.distributedObject = distributedObject;
+	}
+
+	protected Set<String> getCacheEvents() {
+		return cacheEvents;
+	}
+
+	protected DistributedObject getDistributedObject() {
+		return distributedObject;
+	}
```

Comment:
Since `distributedObject` is `final` already this method is redundant and `distributedObject` can become `protected`.